hedera-node/src/main/java/com/hedera/services/store/schedule/HederaScheduleStore.java

```diff
@@ -164,9 +204,44 @@ public ResponseCodeEnum delete(ScheduleID id) {
 	public void commitCreation() {
 		throwIfNoCreationPending();
 
-		var id = fromScheduleId(pendingId);
-		schedules.get().put(id, pendingCreation);
-		extantSchedules.put(pendingCreation.toContentAddressableView(), id);
+		var id = new EntityNumVirtualKey(fromScheduleId(pendingId));
```

Comment:
Lines 207-243 feel like they should be encapsulated in `MerkleScheduledTransactions`.